pkg/gadgets/trace/tcpconnlat/tracer/bpf/tcpconnlat.bpf.c

```diff
@@ -0,0 +1,154 @@
+	__uint(max_entries, 1024);
+	__uint(key_size, sizeof(u64));
+	__uint(value_size, sizeof(u32));
+} mount_ns_filter SEC(".maps");
+
+static int trace_connect(struct sock *sk)
+{
+	u32 tgid = bpf_get_current_pid_tgid() >> 32;
+	struct piddata piddata = {};
+	struct task_struct *task;
+	u64 mntns_id;
+
+	if (targ_tgid && targ_tgid != tgid)
+		return 0;
+
+	task = (struct task_struct*)bpf_get_current_task();
+	mntns_id = (u64) BPF_CORE_READ(task, nsproxy, mnt_ns, ns.inum);
+
+	if (filter_by_mnt_ns && !bpf_map_lookup_elem(&mount_ns_filter, &mntns_id))
+		return 0;
+
+	bpf_get_current_comm(&piddata.comm, sizeof(piddata.comm));
+	piddata.ts = bpf_ktime_get_ns();
+	piddata.tgid = tgid;
+	piddata.mntns_id = mntns_id;
+	bpf_map_update_elem(&start, &sk, &piddata, 0);
+	return 0;
+}
+
+static int handle_tcp_rcv_state_process(void *ctx, struct sock *sk)
+{
+	struct piddata *piddatap;
+	struct event event = {};
+	s64 delta;
+	u64 ts;
+
+	if (BPF_CORE_READ(sk, __sk_common.skc_state) != TCP_SYN_SENT)
+		return 0;
+
+	piddatap = bpf_map_lookup_elem(&start, &sk);
+	if (!piddatap)
+		return 0;
+
+	ts = bpf_ktime_get_ns();
+	delta = (s64)(ts - piddatap->ts);
+	if (delta < 0)
+		goto cleanup;
+
+	event.delta_us = delta / 1000U;
+	if (targ_min_us && event.delta_us < targ_min_us)
+		goto cleanup;
+	__builtin_memcpy(&event.comm, piddatap->comm,
+			sizeof(event.comm));
+	event.ts_us = ts / 1000;
+	event.tgid = piddatap->tgid;
+	event.mntns_id = piddatap->mntns_id;
+	event.lport = BPF_CORE_READ(sk, __sk_common.skc_num);
+	event.dport = BPF_CORE_READ(sk, __sk_common.skc_dport);
+	event.af = BPF_CORE_READ(sk, __sk_common.skc_family);
+	if (event.af == AF_INET) {
+		event.saddr_v4 = BPF_CORE_READ(sk, __sk_common.skc_rcv_saddr);
+		event.daddr_v4 = BPF_CORE_READ(sk, __sk_common.skc_daddr);
+	} else {
+		BPF_CORE_READ_INTO(&event.saddr_v6, sk,
+				__sk_common.skc_v6_rcv_saddr.in6_u.u6_addr32);
+		BPF_CORE_READ_INTO(&event.daddr_v6, sk,
+				__sk_common.skc_v6_daddr.in6_u.u6_addr32);
+	}
+	event.timestamp = bpf_ktime_get_boot_ns();
+	bpf_perf_event_output(ctx, &events, BPF_F_CURRENT_CPU,
+			&event, sizeof(event));
+
+cleanup:
+	bpf_map_delete_elem(&start, &sk);
+	return 0;
+}
+
+SEC("kprobe/tcp_v4_connect")
+int BPF_KPROBE(ig_tcpc_v4_co_e, struct sock *sk)
+{
+	return trace_connect(sk);
+}
+
+SEC("kprobe/tcp_v6_connect")
+int BPF_KPROBE(ig_tcpc_v6_co_e, struct sock *sk)
+{
+	return trace_connect(sk);
+}
+
+SEC("kprobe/tcp_rcv_state_process")
+int BPF_KPROBE(ig_tcp_rsp, struct sock *sk)
+{
+	return handle_tcp_rcv_state_process(ctx, sk);
+}
+
+//SEC("fentry/tcp_v4_connect")
+//int BPF_PROG(fentry_tcp_v4_connect, struct sock *sk)
+//{
+//	return trace_connect(sk);
+//}
+//
+//SEC("fentry/tcp_v6_connect")
+//int BPF_PROG(fentry_tcp_v6_connect, struct sock *sk)
+//{
+//	return trace_connect(sk);
+//}
+//
+//SEC("fentry/tcp_rcv_state_process")
+//int BPF_PROG(fentry_tcp_rcv_state_process, struct sock *sk)
+//{
+//	return handle_tcp_rcv_state_process(ctx, sk);
+//}
```

Comment:
To implement or to remove :)

Reply:
Added reference to https://github.com/inspektor-gadget/inspektor-gadget/issues/1566